apps/server/src/modules/room/api/dto/request/create-room-invitation-link.body.params.ts

```diff
@@ -0,0 +1,53 @@
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { SanitizeHtml } from '@shared/controller/transformer';
+import { IsBoolean, IsDate, IsOptional, IsString, MaxLength, MinLength } from 'class-validator';
+import { RoomInvitationLinkDto } from '../../../domain/do/room-invitation-link.do';
+
+export class CreateRoomInvitationLinkBodyParams
+	implements Omit<RoomInvitationLinkDto, 'creatorUserId' | 'creatorSchoolId'>
+{
+	@ApiProperty({
+		description: 'Id of the room',
+		required: true,
+	})
+	@IsString()
+	roomId!: string;
+
+	@ApiProperty({
+		description: 'Title of the link.',
+	})
+	@MinLength(1)
+	@MaxLength(100)
+	@SanitizeHtml()
+	title!: string;
+
+	@ApiPropertyOptional({
+		description: 'Expiration date of the invitation link',
+		required: false,
+		type: Date,
+	})
+	@IsOptional()
```

Comment:
additional to the `ApiPropertyOptional` really needed?

Reply:
I'm irritated even though we say for some properties that `ApiPropertyOptional` but still we mark them as necessary with "!"